hackathon_site/event/serializers.py

```diff
@@ -85,6 +85,26 @@ def update(self, instance: Profile, validated_data):
         # ProfileSerializer update function and we want to keep the logic between these two update functions separate.
         return serializers.ModelSerializer.update(self, instance, validated_data)
 
+    def create(self, validated_data):
+        try:
+            self.context["request"].user.profile
+            raise serializers.ValidationError("User already has profile")
+        except:
+
+            response_data = {"attended": True}
+            response_data["id_provided"] = False
+            response_data["acknowledge_rules"] = validated_data.pop(
+                "acknowledge_rules", False
+            )
+            response_data["e_signature"] = validated_data.pop("e_signature", "NULL")
+            response_data["user"] = self.context["request"].user
+            response_data["id"] = self.context["request"].user.id
+
+            profile = Profile.objects.create(**response_data)
+            response_data["team"] = self.context["request"].user.profile.team
```

Comment:
why are you setting this after you create the profile?? What's your intention of doing this?

If you look into the Profile model in event/models.py you'll see that if not given a team when created, the Profile model will generate a new team for the profile. Meaning, you don't need it to create the profile but you should set it to `profile.team` not `self.context["request"].user.profile.team`